Add missing metadata when fetching datasets

MatMoore opened the pull request:
I noticed a few bits of metadata are not being passed through to the frontend correctly, so we render some missing values on the details page. This PR adds those in.

We are still missing some attributes we've previously defined like `source_dataset`, `where_to_access_dataset`, but we don't have that in the catalogue yet, so we'll have to handle that in another ticket (either populating or removing these fields).

### Added

- Return `qualifiedName` and `tags` for datasets if available

### Changed

- Use timezone aware timestamps when returning last updated values, and source
  the information from `lastModified` in Datahub, not `lastIngested`.
- Prefer native data types to Datahub's type when returning schemas for datasets

MatMoore committed:
Test tags in dataset response

MatMoore committed:
Return native types if available

When presenting datasets in our UI, use the native data types if we have
them. We suspect these will be more useful to the users.

MatMoore committed:
Return qualifiedName in datasets if available

MatMoore committed:
Return created and modified timestamps

Return created and modified timestamps for datasets and search results.

Previously we were returning the last ingested timestamp for search
results, which is not exactly what we want to present to users. It gives
an indication of whether metadata may be stale or not, but we shouldn't
present that as when the metadata was last modified, or the user will
conclude that there has been some change that never happened.

Returning this for datasets as well means we can display it on the
details page.

MatMoore committed:
Bump version to 0.23.0

MatMoore committed:
Lint

LavMatt commented on python-libraries/data-platform-catalogue/data_platform_catalogue/entities.py:
```diff
@@ -121,13 +121,15 @@ class TableMetadata:
     owner: str = ""
     owner_email: str = ""
+    last_updated: datetime | None = None
+    first_created: datetime | None = None
```
Not necessarily for this PR (could be a separate ticket) but i think we should start adding descriptions, using pydantic's `Field` possibly, to our entity metadata properties. They are growing quickly and some are a little bit open to interpretation, e.g. `first_created` could mean first created in the catalogue or first created as a table on the AP, or even when the table first was created at source

MatMoore replied:
Yeah 💯 I would be keen to redo all these model classes.

I'd be in favour of adopting pydantic for this, as then we can serialise/deserialise to JSON as well without additional code.

LavMatt commented on python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/datahub_client.py:
```diff
@@ -664,6 +665,8 @@ def get_table_details(self, urn) -> TableMetadata:
             domain = parse_domain(response)
             owner, owner_email = parse_owner(response)
             tags = parse_tags(response)
+            name = properties["name"]
```
This is throwing an `KeyError`when run for dbt dataset urn. It's something that has slipped through,`name` is not set within the `properties` key of the response for dbt ingestions. They have a `name` key/value at the top level of the response, however this is equivalent to a fully qualified name e.g. `awsdatacatalog.dbt_database.dbt_table`

MatMoore replied:
Ah, annoying. I've added a fallback to top level `name` to match the search response.

LavMatt reviewed (commented):
Looks good. I've picked up one issue that is unrelated to the changes here and a comment on descriptions for our entity metadata properties. 

Could be handled in fresh tickets if preferable

MatMoore committed:
Fall back to top level name attribute if missing from properties

According to https://datahubproject.io/docs/graphql/objects/#dataset
name is not supposed to be used anymore, but it seems like the dbt
ingestion is populating this instead of properties.name.
